beacon-chain/blockchain/core.go

```diff
@@ -490,22 +473,126 @@ func (b *BeaconChain) validatorsByHeightShard() ([]*beaconCommittee, error) {
 	return committees, nil
 }
 
-// GetIndicesForHeight returns the attester set of a given height.
-func (b *BeaconChain) GetIndicesForHeight(height uint64) (*pb.ShardAndCommitteeArray, error) {
+// getIndicesForSlot returns the attester set of a given slot.
+func (b *BeaconChain) getIndicesForSlot(slot uint64) (*pb.ShardAndCommitteeArray, error) {
 	lcs := b.CrystallizedState().LastStateRecalc()
-	if !(lcs <= height && height < lcs+params.CycleLength*2) {
-		return nil, fmt.Errorf("can not return attester set of given height, input height %v has to be in between %v and %v", height, lcs, lcs+params.CycleLength*2)
+	if !(lcs <= slot && slot < lcs+params.CycleLength*2) {
+		return nil, fmt.Errorf("can not return attester set of given slot, input slot %v has to be in between %v and %v", slot, lcs, lcs+params.CycleLength*2)
 	}
-	return b.CrystallizedState().IndicesForHeights()[height-lcs], nil
+	return b.CrystallizedState().IndicesForHeights()[slot-lcs], nil
 }
 
-// GetBlockHash returns the block hash of a given height.
-func (b *BeaconChain) GetBlockHash(slot, height uint64) ([]byte, error) {
-	sback := slot - params.CycleLength*2
-	if !(sback <= height && height < sback+params.CycleLength*2) {
-		return nil, fmt.Errorf("can not return attester set of given height, input height %v has to be in between %v and %v", height, sback, sback+params.CycleLength*2)
+// getBlockHash returns the block hash of a slot.
+func (b *BeaconChain) getBlockHash(slot uint64, block *types.Block) ([]byte, error) {
+	sback := int(block.SlotNumber()) - params.CycleLength*2
```

Comment:
slot back. `getBlockHash` can only return within a range of `block.Slot - 128 <= slot < block.Slot`.  Let's say current block is 200, but you wanna get the block hash of slot 50. That won't work because the further you can get is 200 - 128. And that is slot back

Reply:
Will change `sback` to `slotBack`